upgrade to latest sagemaker-experiments

danabens opened the pull request:
upgrade to latest sagemaker-experiments

codewithsk commented:
Can you rebase this branch with the master to get the latest changes?

laurenyu commented on test/integration/conftest.py:
```diff
@@ -116,3 +116,10 @@ def docker_image(docker_base_name, tag):
+
+
+@pytest.fixture(autouse=True)
+def skip_py2_containers(request, tag):
```
there is a fixture for the Python version: https://github.com/aws/sagemaker-tensorflow-container/blob/master/test/integration/utils.py#L30-L32

if you want to use it, you could define this `skip_py2_containers` fixture to the `test_experiments` file

danabens replied:
removed the fixture

laurenyu commented:
here's the latest error I see in the PR build:

```
==================================== ERRORS ====================================
_______________________ ERROR at setup of test_training ________________________
[gw0] linux -- Python 3.6.9 /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/.tox/py36/bin/python
file /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/test/integration/sagemaker/test_experiments.py, line 32
  @pytest.mark.skip_py2_containers
  def test_training(sagemaker_session, ecr_image, instance_type, framework_version):
file /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py, line 95
  @pytest.fixture(autouse=True)
  def skip_by_device_type(request, processor):
E       fixture 'processor' not found
>       available fixtures: account_id, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, docker_base_name, docker_image, doctest_namespace, ecr_image, framework_version, instance_type, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, region, requests_mock, sagemaker_local_session, sagemaker_session, skip_by_device_type, skip_gpu_instance_restricted_regions, skip_py2_containers, tag, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py:95
```

I think you might need to do something like this: https://github.com/aws/sagemaker-tensorflow-container/blob/master/test/integration/local/test_horovod.py#L22

laurenyu commented on test/integration/sagemaker/test_experiments.py:
```diff
@@ -0,0 +1,98 @@
+SCRIPT_PATH = os.path.join(DATA_PATH, "mnist_gluon_basic_hook_demo.py")
+
+
+def test_training(sagemaker_session, ecr_image, instance_type, framework_version, py_version):
```
the last PR build had [the same failure as before](https://github.com/aws/sagemaker-tensorflow-container/pull/286#issuecomment-595884123). I think you need to do something similar to https://github.com/aws/sagemaker-tensorflow-container/blob/master/test/integration/local/test_horovod.py#L22 and put this line in this file:

```python
from test.integration.utils import processor, py_version  # noqa: F401
```

danabens commented:
> here's the latest error I see in the PR build:
> 
> ```
> ==================================== ERRORS ====================================
> _______________________ ERROR at setup of test_training ________________________
> [gw0] linux -- Python 3.6.9 /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/.tox/py36/bin/python
> file /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/test/integration/sagemaker/test_experiments.py, line 32
>   @pytest.mark.skip_py2_containers
>   def test_training(sagemaker_session, ecr_image, instance_type, framework_version):
> file /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py, line 95
>   @pytest.fixture(autouse=True)
>   def skip_by_device_type(request, processor):
> E       fixture 'processor' not found
> >       available fixtures: account_id, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, docker_base_name, docker_image, doctest_namespace, ecr_image, framework_version, instance_type, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, region, requests_mock, sagemaker_local_session, sagemaker_session, skip_by_device_type, skip_gpu_instance_restricted_regions, skip_py2_containers, tag, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
> >       use 'pytest --fixtures [testpath]' for help on them.
> 
> /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py:95
> ```
> 
> I think you might need to do something like this: https://github.com/aws/sagemaker-tensorflow-container/blob/master/test/integration/local/test_horovod.py#L22

I removed the fixture i added but still seeing  ```fixture 'processor' not found```

laurenyu commented:
> > here's the latest error I see in the PR build:
> > ```
> > ==================================== ERRORS ====================================
> > _______________________ ERROR at setup of test_training ________________________
> > [gw0] linux -- Python 3.6.9 /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/.tox/py36/bin/python
> > file /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/test/integration/sagemaker/test_experiments.py, line 32
> >   @pytest.mark.skip_py2_containers
> >   def test_training(sagemaker_session, ecr_image, instance_type, framework_version):
> > file /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py, line 95
> >   @pytest.fixture(autouse=True)
> >   def skip_by_device_type(request, processor):
> > E       fixture 'processor' not found
> > >       available fixtures: account_id, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, docker_base_name, docker_image, doctest_namespace, ecr_image, framework_version, instance_type, monkeypatch, no_cover, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, region, requests_mock, sagemaker_local_session, sagemaker_session, skip_by_device_type, skip_gpu_instance_restricted_regions, skip_py2_containers, tag, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
> > >       use 'pytest --fixtures [testpath]' for help on them.
> > 
> > /codebuild/output/src286481858/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py:95
> > ```
> > 
> > 
> > I think you might need to do something like this: https://github.com/aws/sagemaker-tensorflow-container/blob/master/test/integration/local/test_horovod.py#L22
> 
> I removed the fixture i added but still seeing `fixture 'processor' not found`

I think you need to specifically add this line:

```python
from test.integration.utils import processor, py_version  # noqa: F401
```

laurenyu commented:
latest error was due to the flake8 check:

```
flake8 run-test: commands[0] | flake8
./test/integration/sagemaker/test_experiments.py:26:1: F401 'test.integration.utils.processor' imported but unused
ERROR: InvocationError for command /codebuild/output/src000776859/src/github.com/aws/sagemaker-tensorflow-container/.tox/flake8/bin/flake8 (exited with code 1)
```

danabens commented:
> latest error was due to the flake8 check:
> 
> ```
> flake8 run-test: commands[0] | flake8
> ./test/integration/sagemaker/test_experiments.py:26:1: F401 'test.integration.utils.processor' imported but unused
> ERROR: InvocationError for command /codebuild/output/src000776859/src/github.com/aws/sagemaker-tensorflow-container/.tox/flake8/bin/flake8 (exited with code 1)
> ```
weird, i ran flake8 before updating. investigating.
```
    danabens   ~/w/g/sagemaker-tensorflow-c     experiments-upgrade   2                                                                                                        1 ↵  10:35 
╰ tox -e flake8
zsh: correct 'flake8' to '.flake8' [nyae]? n
GLOB sdist-make: /Users/danabens/workplace/github/sagemaker-tensorflow-container/setup.py
flake8 inst-nodeps: /Users/danabens/workplace/github/sagemaker-tensorflow-container/.tox/.tmp/package/1/sagemaker_tensorflow_training-2.2.3.dev0.zip
flake8 installed: entrypoints==0.3,flake8==3.7.9,flake8-future-import==0.4.6,flake8-import-order==0.18.1,mccabe==0.6.1,pycodestyle==2.5.0,pyflakes==2.1.1,sagemaker-tensorflow-training==2.2.3.dev0
flake8 run-test-pre: PYTHONHASHSEED='1745302340'
flake8 run-test: commands[0] | flake8
________________________________________________________________________________________________ summary _________________________________________________________________________________________________
  flake8: commands succeeded
  congratulations :)
```

codewithsk commented:
```
ImportError while loading conftest '/codebuild/output/src238486594/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py'.
test/integration/conftest.py:95: in <module>
    @pytest.fixture(params=os.environ['TEST_PROCESSORS'].split(','))
.tox/py36/lib/python3.6/os.py:669: in __getitem__
    raise KeyError(key) from None
E   KeyError: 'TEST_PROCESSORS'
Coverage.py warning: Module sagemaker_tensorflow_container was never imported. (module-not-imported)
Coverage.py warning: No data was collected. (no-data-collected)
ERROR: InvocationError for command /codebuild/output/src238486594/src/github.com/aws/sagemaker-tensorflow-container/.tox/py36/bin/coverage run --rcfile .coveragerc_py36 --source sagemaker_tensorflow_container -m py.test test/integration/local --region us-west-2 --docker-base-name 142577830533.dkr.ecr.us-west-2.amazonaws.com/sagemaker-test --tag 1.15.2-cpu-py3-sagemaker-tensorflow-container-pr-ce5c15ed-a05a-4d22-ae7d-1c7b807f6461 --framework-version 1.15.2 --py-version 3 --processor cpu --durations 10 (exited with code 4)
```

danabens commented:
```
==================================== ERRORS ====================================
_______________________ ERROR at setup of test_training ________________________
[gw1] linux -- Python 3.6.9 /codebuild/output/src112555631/src/github.com/aws/sagemaker-tensorflow-container/.tox/py36/bin/python
file /codebuild/output/src112555631/src/github.com/aws/sagemaker-tensorflow-container/test/integration/sagemaker/test_experiments.py, line 32
  def test_training(sagemaker_session, ecr_image, instance_type, framework_version, py_version):
file /codebuild/output/src112555631/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py, line 122
  @pytest.fixture
  def ecr_image(account_id, docker_base_name, tag, region):
file /codebuild/output/src112555631/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py, line 66
  @pytest.fixture
  def tag(request, framework_version, processor, py_version):
E       fixture 'py_version' not found
>       available fixtures: account_id, cache, capfd, capfdbinary, caplog, capsys, capsysbinary, cov, docker_base_name, docker_image, doctest_namespace, ecr_image, framework_version, instance_type, monkeypatch, no_cover, processor, pytestconfig, record_property, record_testsuite_property, record_xml_attribute, recwarn, region, requests_mock, sagemaker_local_session, sagemaker_session, skip_by_device_type, skip_gpu_instance_restricted_regions, tag, tmp_path, tmp_path_factory, tmpdir, tmpdir_factory, worker_id
>       use 'pytest --fixtures [testpath]' for help on them.

/codebuild/output/src112555631/src/github.com/aws/sagemaker-tensorflow-container/test/integration/conftest.py:66
```

codewithsk commented:
```
self = <botocore.validate.ParamValidationDecorator object at 0x7fe8e17287b8>
parameters = {'TrialName': 'tf-container-integ-test-1583792568'}
operation_model = OperationModel(name=ListTrialComponents)

    def serialize_to_request(self, parameters, operation_model):
        input_shape = operation_model.input_shape
        if input_shape is not None:
            report = self._param_validator.validate(parameters,
                                                    operation_model.input_shape)
            if report.has_errors():
>               raise ParamValidationError(report=report.generate_report())
E               botocore.exceptions.ParamValidationError: Parameter validation failed:
E               Unknown parameter in input: "TrialName", must be one of: SourceArn, CreatedAfter, CreatedBefore, SortBy, SortOrder, MaxResults, NextToken
```

danabens commented:
i rebased onto latest master and get:
```
Removing intermediate container bdd44c2df758
 ---> 70ad5bf0207d
Step 26/31 : COPY $FRAMEWORK_SUPPORT_INSTALLABLE .
 ---> af15e3cad781
Step 27/31 : RUN pip3 --no-cache-dir install --upgrade     pip     setuptools
 ---> Running in 0bde5e70336c
Collecting pip
  Downloading https://files.pythonhosted.org/packages/54/0c/d01aa759fdc501a58f431eb594a17495f15b88da142ce14b5845662c13f3/pip-20.0.2-py2.py3-none-any.whl (1.4MB)
Collecting setuptools
  Downloading https://files.pythonhosted.org/packages/70/b8/b23170ddda9f07c3444d49accde49f2b92f97bb2f2ebc312618ef12e4bd6/setuptools-46.0.0-py3-none-any.whl (582kB)
Installing collected packages: pip, setuptools
  Found existing installation: pip 9.0.1
    Not uninstalling pip at /usr/lib/python3/dist-packages, outside environment /usr
  Found existing installation: setuptools 39.0.1
    Not uninstalling setuptools at /usr/lib/python3/dist-packages, outside environment /usr
Successfully installed pip-20.0.2 setuptools-46.0.0
Removing intermediate container 0bde5e70336c
 ---> d59ae273c3a0
Step 28/31 : RUN ln -s $(which python3) /usr/local/bin/python  && ln -s $(which pip3) /usr/bin/pip
 ---> Running in 8a70d17bab6f
Removing intermediate container 8a70d17bab6f
 ---> 39cad4ef9531
Step 29/31 : RUN pip install --no-cache-dir -U     numpy==1.17.4     scipy==1.2.2     scikit-learn==0.20.3     pandas==0.24.2     Pillow==6.2.1     h5py==2.9.0     keras_applications==1.0.8     keras_preprocessing==1.1.0     keras==2.3.1     requests==2.22.0     ${SMDEBUG_URL}     sagemaker==1.50.17     sagemaker-experiments==0.1.7     mpi4py==3.0.2     "cryptography>=2.3"     "sagemaker-tensorflow>=1.15,<1.16"  && pip install --force-reinstall --no-cache-dir -U     ${TF_URL}  && pip install --force-reinstall --no-cache-dir -U     horovod==0.18.2  && pip install --no-cache-dir -U     $FRAMEWORK_SUPPORT_INSTALLABLE     awscli && rm -f $FRAMEWORK_SUPPORT_INSTALLABLE
 ---> Running in a23aa88e9147
Collecting numpy==1.17.4
  Downloading numpy-1.17.4-cp36-cp36m-manylinux1_x86_64.whl (20.0 MB)
Collecting scipy==1.2.2
  Downloading scipy-1.2.2-cp36-cp36m-manylinux1_x86_64.whl (24.8 MB)
Collecting scikit-learn==0.20.3
  Downloading scikit_learn-0.20.3-cp36-cp36m-manylinux1_x86_64.whl (5.4 MB)
Collecting pandas==0.24.2
  Downloading pandas-0.24.2-cp36-cp36m-manylinux1_x86_64.whl (10.1 MB)
Collecting Pillow==6.2.1
  Downloading Pillow-6.2.1-cp36-cp36m-manylinux1_x86_64.whl (2.1 MB)
Collecting h5py==2.9.0
  Downloading h5py-2.9.0-cp36-cp36m-manylinux1_x86_64.whl (2.8 MB)
Collecting keras_applications==1.0.8
  Downloading Keras_Applications-1.0.8-py3-none-any.whl (50 kB)
Collecting keras_preprocessing==1.1.0
  Downloading Keras_Preprocessing-1.1.0-py2.py3-none-any.whl (41 kB)
Collecting keras==2.3.1
  Downloading Keras-2.3.1-py2.py3-none-any.whl (377 kB)
Collecting requests==2.22.0
  Downloading requests-2.22.0-py2.py3-none-any.whl (57 kB)
Collecting smdebug==0.7.0b20200304
[91m  ERROR: HTTP error 403 while getting https://s3.amazonaws.com/kannanva-smdebug-0.7/smdebug-0.7.0b20200304-py2.py3-none-any.whl
[0m[91m  ERROR: Could not install requirement smdebug==0.7.0b20200304 from https://s3.amazonaws.com/kannanva-smdebug-0.7/smdebug-0.7.0b20200304-py2.py3-none-any.whl because of error 403 Client Error: Forbidden for url: https://s3.amazonaws.com/kannanva-smdebug-0.7/smdebug-0.7.0b20200304-py2.py3-none-any.whl
[0m[91mERROR: Could not install requirement smdebug==0.7.0b20200304 from https://s3.amazonaws.com/kannanva-smdebug-0.7/smdebug-0.7.0b20200304-py2.py3-none-any.whl because of HTTP error 403 Client Error: Forbidden for url: https://s3.amazonaws.com/kannanva-smdebug-0.7/smdebug-0.7.0b20200304-py2.py3-none-any.whl for URL https://s3.amazonaws.com/kannanva-smdebug-0.7/smdebug-0.7.0b20200304-py2.py3-none-any.whl
[0mThe command '/bin/sh -c pip install --no-cache-dir -U     numpy==1.17.4     scipy==1.2.2     scikit-learn==0.20.3     pandas==0.24.2     Pillow==6.2.1     h5py==2.9.0     keras_applications==1.0.8     keras_preprocessing==1.1.0     keras==2.3.1     requests==2.22.0     ${SMDEBUG_URL}     sagemaker==1.50.17     sagemaker-experiments==0.1.7     mpi4py==3.0.2     "cryptography>=2.3"     "sagemaker-tensorflow>=1.15,<1.16"  && pip install --force-reinstall --no-cache-dir -U     ${TF_URL}  && pip install --force-reinstall --no-cache-dir -U     horovod==0.18.2  && pip install --no-cache-dir -U     $FRAMEWORK_SUPPORT_INSTALLABLE     awscli && rm -f $FRAMEWORK_SUPPORT_INSTALLABLE' returned a non-zero code: 1
```

codewithsk commented:
```
==================================== ERRORS ====================================
_______ ERROR collecting test/integration/sagemaker/test_experiments.py ________
ImportError while importing test module '/codebuild/output/src627080960/src/github.com/aws/sagemaker-tensorflow-container/test/integration/sagemaker/test_experiments.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
test/integration/sagemaker/test_experiments.py:21: in <module>
    from smexperiments.experiment import Experiment
E   ImportError: No module named smexperiments.experiment
```

danabens commented:
the test has the "skip_py2_containers" mark on on it but is still being run with the py2 target:
```
No data to report.
___________________________________ summary ____________________________________
  py36: commands succeeded
  congratulations :)

[Container] 2020/03/10 19:49:05 Running command test_cmd="IGNORE_COVERAGE=- tox -e py27 -- test/integration/sagemaker -n 8 --region $AWS_DEFAULT_REGION --docker-base-name $ECR_REPO --account-id $ACCOUNT --tag $CPU_TAG_PY2 --py-version 2 --processor cpu --durations 10"

[Container] 2020/03/10 19:49:05 Running command execute-command-if-has-matching-changes "$test_cmd" "test/" "src/*.py" "setup.py" "docker/*" "buildspec.yml"
Changes Found. Executing command passed in as first parameter.
GLOB sdist-make: /codebuild/output/src627080960/src/github.com/aws/sagemaker-tensorflow-container/setup.py
py27 inst-nodeps: /codebuild/output/src627080960/src/github.com/aws/sagemaker-tensorflow-container/.tox/.tmp/package/1/sagemaker_tensorflow_training-2.2.4.dev0.zip
py27 installed: DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support,absl-py==0.9.0,apipkg==1.5,appdirs==1.4.3,astor==0.8.1,atomicwrites==1.3.0,attrs==19.3.0,awscli==1.16.314,backports.shutil-get-terminal-size==1.0.0,backports.ssl-match-hostname==3.7.0.1,backports.weakref==1.0.post1,bcrypt==3.1.7,boto3==1.10.50,botocore==1.13.50,cached-property==1.5.1,certifi==2019.11.28,cffi==1.14.0,chardet==3.0.4,click==7.1.1,colorama==0.4.1,configparser==4.0.2,contextlib2==0.6.0.post1,coverage==5.0.3,cryptography==2.8,distlib==0.3.0,docker==4.2.0,docker-compose==1.25.4,dockerpty==0.4.1,docopt==0.6.2,docutils==0.15.2,entrypoints==0.3,enum34==1.1.10,execnet==1.7.1,filelock==3.0.12,flake8==3.7.9,Flask==1.1.1,funcsigs==1.0.2,functools32==3.2.3.post2,futures==3.3.0,gast==0.2.2,gevent==1.4.0,google-pasta==0.1.8,greenlet==0.4.15,grpcio==1.27.2,gunicorn==19.10.0,h5py==2.10.0,idna==2.9,importlib-metadata==1.5.0,importlib-resources==1.3.1,inotify-simple==1.2.1,ipaddress==1.0.23,itsdangerous==1.1.0,Jinja2==2.11.1,jmespath==0.9.5,jsonschema==3.2.0,Keras-Applications==1.0.8,Keras-Preprocessing==1.1.0,Markdown==3.1.1,MarkupSafe==1.1.1,mccabe==0.6.1,mock==3.0.5,more-itertools==5.0.0,numpy==1.16.6,opt-einsum==2.3.2,packaging==20.3,pandas==0.24.2,paramiko==2.4.2,pathlib2==2.3.5,Pillow==6.2.2,pluggy==0.13.1,protobuf==3.11.3,protobuf3-to-dict==0.1.5,psutil==5.4.8,py==1.8.1,pyasn1==0.4.8,pycodestyle==2.5.0,pycparser==2.20,pyflakes==2.1.1,PyNaCl==1.3.0,pyparsing==2.4.6,pyrsistent==0.15.7,pytest==4.6.9,pytest-cov==2.8.1,pytest-forked==1.1.3,pytest-xdist==1.31.0,python-dateutil==2.8.0,pytz==2019.3,PyYAML==5.3,requests==2.23.0,requests-mock==1.7.0,retrying==1.3.3,rsa==3.4.2,s3transfer==0.2.1,sagemaker==1.50.1,sagemaker-containers==2.8.1,sagemaker-tensorflow-training==2.2.4.dev0,scandir==1.10.0,scikit-learn==0.20.4,scipy==1.2.3,singledispatch==3.4.0.3,six==1.13.0,sklearn==0.0,smdebug-rulesconfig==0.1.2,subprocess32==3.5.4,tensorboard==1.15.0,tensorflow==1.15.0,tensorflow-estimator==1.15.1,termcolor==1.1.0,texttable==1.6.2,toml==0.10.0,tox==3.14.5,typing==3.7.4.1,urllib3==1.25.8,virtualenv==20.0.10,wcwidth==0.1.8,websocket-client==0.57.0,Werkzeug==0.15.5,wrapt==1.12.1,zipp==1.2.0
py27 run-test-pre: PYTHONHASHSEED='4200421712'
py27 run-test: commands[0] | coverage run --rcfile .coveragerc_py27 --source sagemaker_tensorflow_container -m py.test test/integration/sagemaker -n 8 --region us-west-2 --docker-base-name sagemaker-test --account-id 142577830533 --tag 1.15.2-cpu-py2-sagemaker-tensorflow-container-pr-8789f131-d767-49bd-97a5-51c297ed6f8e --py-version 2 --processor cpu --durations 10
============================= test session starts ==============================
platform linux2 -- Python 2.7.17, pytest-4.6.9, py-1.8.1, pluggy-0.13.1
cachedir: .tox/py27/.pytest_cache
rootdir: /codebuild/output/src627080960/src/github.com/aws/sagemaker-tensorflow-container, inifile: pytest.ini
plugins: xdist-1.31.0, cov-2.8.1, forked-1.1.3, requests-mock-1.7.0
gw0 I / gw1 I / gw2 I / gw3 I / gw4 I / gw5 I / gw6 I / gw7 I
gw0 [7] / gw1 [7] / gw2 [7] / gw3 [7] / gw4 [7] / gw5 [7] / gw6 [7] / gw7 [7]

.....F.                                                                  [100%]
==================================== ERRORS ====================================
_______ ERROR collecting test/integration/sagemaker/test_experiments.py ________
ImportError while importing test module '/codebuild/output/src627080960/src/github.com/aws/sagemaker-tensorflow-container/test/integration/sagemaker/test_experiments.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
test/integration/sagemaker/test_experiments.py:21: in <module>
    from smexperiments.experiment import Experiment
E   ImportError: No module named smexperiments.experiment
=================================== FAILURES ===================================
```

codewithsk commented on test/integration/sagemaker/test_experiments.py:
```diff
@@ -0,0 +1,95 @@
+from sagemaker.tensorflow import TensorFlow
+from smexperiments.experiment import Experiment
+from smexperiments.trial import Trial
+from smexperiments.trial_component import TrialComponent
```
Can you move these import lines to be inside the function? After line 36?

danabens replied:
ya sure

codewithsk commented:
@danabens Build still failling. Can you try a pytest fixture like this: https://github.com/aws/sagemaker-mxnet-container/blob/6a1df9164b0c94764f4ace3ef3310ac4fc2e8fb9/test/conftest.py#L134 ?

codewithsk commented:
```
=================================== FAILURES ===================================
________________________________ test_training _________________________________
[gw5] linux2 -- Python 2.7.17 /codebuild/output/src003255308/src/github.com/aws/sagemaker-tensorflow-container/.tox/py27/bin/python

sagemaker_session = <sagemaker.session.Session object at 0x7fd651564ad0>
ecr_image = '142577830533.dkr.ecr.us-west-2.amazonaws.com/sagemaker-test:1.15.2-cpu-py2-sagemaker-tensorflow-container-pr-f0586a4f-1e26-481c-8499-2e169a685adb'
instance_type = 'ml.p2.xlarge', framework_version = '2.0.0', py_version = ['2']

    @pytest.mark.skip_py2_containers
    def test_training(sagemaker_session, ecr_image, instance_type, framework_version, py_version):
    
        if py_version == '2' or py_version is None:
            pytest.skip('Skipping python2 {}'.format(py_version))
            return
    
>       from smexperiments.experiment import Experiment
E       ImportError: No module named smexperiments.experiment

test/integration/sagemaker/test_experiments.py:36: ImportError
=============================== warnings summary ===============================
.tox/py27/local/lib/python2.7/site-packages/_pytest/mark/structures.py:335
.tox/py27/local/lib/python2.7/site-packages/_pytest/mark/structures.py:335
.tox/py27/local/lib/python2.7/site-packages/_pytest/mark/structures.py:335
.tox/py27/local/lib/python2.7/site-packages/_pytest/mark/structures.py:335
.tox/py27/local/lib/python2.7/site-packages/_pytest/mark/structures.py:335
.tox/py27/local/lib/python2.7/site-packages/_pytest/mark/structures.py:335
.tox/py27/local/lib/python2.7/site-packages/_pytest/mark/structures.py:335
.tox/py27/local/lib/python2.7/site-packages/_pytest/mark/structures.py:335
  /codebuild/output/src003255308/src/github.com/aws/sagemaker-tensorflow-container/.tox/py27/local/lib/python2.7/site-packages/_pytest/mark/structures.py:335: PytestUnknownMarkWarning: Unknown pytest.mark.skip_py2_containers - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
```

danabens commented:
```
========================== slowest 10 test durations ===========================
118.05s call     test/integration/local/test_training.py::test_gpu[gpu-2]
0.03s setup    test/integration/local/test_horovod.py::test_distributed_training_horovod_basic[gpu-2-5-2]
0.03s setup    test/integration/local/test_horovod.py::test_distributed_training_horovod_basic[gpu-2-2-2]
0.03s setup    test/integration/local/test_horovod.py::test_distributed_training_horovod_basic[gpu-2-2-1]
0.02s setup    test/integration/local/test_horovod.py::test_distributed_training_horovod_basic[gpu-2-1-2]

(0.00 durations hidden.  Use -vv to show these durations.)
==================== 1 passed, 9 skipped in 120.01 seconds =====================
Coverage.py warning: Module sagemaker_tensorflow_container was never imported. (module-not-imported)
Coverage.py warning: No data was collected. (no-data-collected)
py27 run-test: commands[1] | - coverage report --include '*sagemaker_tensorflow_container*' --show-missing
No data to report.
___________________________________ summary ____________________________________
  py27: commands succeeded
  congratulations :)
```

codewithsk commented on setup.py:
```diff
@@ -60,6 +60,7 @@ def read_version():
                  'requests-mock', 'awscli==1.16.314'],
-        'benchmark': ['click']
+        'benchmark': ['click'],
+        ':python_version=="3.6"': ['sagemaker-experiments==0.1.7']
```
Would this need to be fixed to be >3.5 too?

codewithsk replied:
Just like in the MXNet/Pytorch containers?

danabens replied:
it will work for now since the only 3 version is 3.6. should be updated it for future revisions though.

danabens replied:
ill update tf once this pr is merged
